platform/source/minimal-printf/mbed_printf_implementation.c

```diff
@@ -194,17 +194,23 @@ static void mbed_minimal_formatted_string_unsigned(char *buffer, size_t length,
 /**
  * @brief      Print hexadecimal.
  *
- * @param      buffer  The buffer to store output (NULL for stdout).
- * @param[in]  length  The length of the buffer.
- * @param      result  The current output location.
- * @param[in]  value   The value to be printed.
- * @param      upper   Flag to print the hexadecimal in upper or lower case.
+ * @param      buffer           The buffer to store output (NULL for stdout).
+ * @param[in]  length           The length of the buffer.
+ * @param      result           The current output location.
+ * @param[in]  value            The value to be printed.
+ * @param      upper            Flag to print the hexadecimal in upper or lower case.
+ * @param      width_modifier   The width modifier. Only even values are supported. (0 equals 16, 16 max)
+ * @param      prepend_zeros    Flag to prepends zeros when the width_modifier is greater than 0
  */
-static void mbed_minimal_formatted_string_hexadecimal(char *buffer, size_t length, int *result, MBED_UNSIGNED_STORAGE value, FILE *stream, bool upper)
+static void mbed_minimal_formatted_string_hexadecimal(char *buffer, size_t length, int *result, MBED_UNSIGNED_STORAGE value, bool upper, int width_modifier, bool prepend_zeros, FILE *stream)
```

Comment:
```suggestion
static void mbed_minimal_formatted_string_hexadecimal(
    char *buffer,
    size_t length,
    int *result,
    MBED_UNSIGNED_STORAGE value,
    bool upper,
    int width_modifier,
    bool prepend_zeros,
    FILE *stream
)
```
It is getting a little bit too wide.